src/Lecture2.hs

```diff
@@ -52,7 +52,10 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct [] = 1
+lazyProduct [x] = x
```

Comment:
This could be a nice potential optimization but semantically this line is not needed 😌